### PR TITLE
Replace nav-link with util-nav-link-attr

### PR DESCRIPTION
**Pull request by dominic-blain**
Proposal to convert nav link to attributes only instead of full button template. Would complement [this pull request](https://github.com/DeuxHuitHuit/framework.js-modules/pull/201)

### CODE REVIEW

**Comment by fhamon** [comment]
I like the idea. Should we call it by default in lib/button.xsl ?

**Comment by hotdoy** [comment]
10/10

**Comment by dominic-blain** [comment]
> I like the idea. Should we call it by default in lib/button.xsl ?

Don't think so.. since its for nav buttons only?

**Comment by fhamon** [comment]
Oh I thought that was for all buttons. héhé. Either way, LGTM.

File: workspace/utilities/lib/util-nav-link-attr.xsl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		<xsl:param name="target" select="'_self'" />
+		<xsl:param name="ext-attr" />
+
+		<xsl:variable name="has-ga-cat" select="string-length($ga-cat) != 0" />
+		<xsl:variable name="has-ga-label" select="string-length($ga-label) != 0" />
+		<xsl:variable name="has-role" select="string-length($role) != 0" />
+		<xsl:variable name="has-target" select="string-length($target) != 0" />
+
+		<xsl:if test="$has-ga-cat">
+			<add data-ga-cat="{$ga-cat}" />
```
Comment (PascalPiche) [inline workspace/utilities/lib/util-nav-link-attr.xsl]: use set instead of add

File: workspace/utilities/lib/util-nav-link-attr.xsl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		</xsl:if>
+		<xsl:if test="$has-ga-label">
+			<add data-ga-label="{$ga-label}" />
```
Comment (PascalPiche) [inline workspace/utilities/lib/util-nav-link-attr.xsl]: use set instead of add

File: workspace/utilities/lib/util-nav-link-attr.xsl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		</xsl:if>
+		<xsl:if test="$has-role">
+			<add role="{$role}" />
```
Comment (PascalPiche) [inline workspace/utilities/lib/util-nav-link-attr.xsl]: use set instead of add

File: workspace/utilities/lib/util-nav-link-attr.xsl [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		</xsl:if>
+		<xsl:if test="$has-target">
+			<add target="{$target}" />
```
Comment (PascalPiche) [inline workspace/utilities/lib/util-nav-link-attr.xsl]: use set instead of add

File: workspace/utilities/lib/util-nav-link-attr.xsl [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+<!-- COMPONENT: util-nav-link-attr -->
+	<xsl:template name="util-nav-link-attr">
+		<xsl:param name="label" select="label" />
+		<xsl:param name="ga-cat" select="'menu-nav'" />
+		<xsl:param name="ga-label" select="concat($url-language, '/', $label)" />
+		<xsl:param name="role" select="'menuitem'" />
+		<xsl:param name="target" select="'_self'" />
+		<xsl:param name="ext-attr" />
```
Comment (PascalPiche) [inline workspace/utilities/lib/util-nav-link-attr.xsl]: not necessary

Comment (dominic-blain) [reply workspace/utilities/lib/util-nav-link-attr.xsl]: But can it be someday? Other than that it's just cleaner to have all values up top don't you think?

Comment (dominic-blain) [reply workspace/utilities/lib/util-nav-link-attr.xsl]: OOOOOOhhhh, nevermind, didn't understood you were talking about **ext-attr** only.

**Comment by dominic-blain** [comment]
@PascalPiche Fixed

**Comment by dominic-blain** [comment]
@nitriques Mmm.. Yes will update but it's also used in site-nav. I would like to simply remove the call to nav-link template in site-nav and update site-nav in another template to keep PR simple. Good with you? (site-nav always gets rewritten anyway)

**Comment by dominic-blain** [comment]
Ping @nitriques

File: workspace/utilities/lib/util-nav-link-attr.xsl (hunk 1 above)
Comment (nitriques) [reply workspace/utilities/lib/util-nav-link-attr.xsl]: @dominic-blain ^^

**Comment by nitriques** [comment]
Also, it needs to be rebased...

File: workspace/utilities/lib/util-nav-link-attr.xsl (hunk 1 above)
Comment (dominic-blain) [reply workspace/utilities/lib/util-nav-link-attr.xsl]: @nitriques Outdated, was fixed 8 days ago!

**Comment by dominic-blain** [comment]
@nitriques 
> Also, it needs to be rebased...
Done!
> util-nav-link-attr should be called by default. Maybe in computed attr ?
Where? In site-nav? I still think I should only touch site-nav in another PR.

File: workspace/utilities/master/master.xsl [hunk 6]
```diff
@@ -36,7 +36,7 @@
 	<!-- Buttons -->
 	<xsl:import href="../lib/button.xsl" />
 	<xsl:import href="../lib/button-tel.xsl" />
-	<xsl:import href="../lib/nav-link.xsl" />
+	<xsl:import href="../lib/edit-buttons.xsl" />
```
Comment (nitriques) [inline workspace/utilities/master/master.xsl]: This should be another PR.. or maybe we now have it in duplicate ??

**Comment by nitriques** [comment]
@dominic-blain site-nav needs to enforce it somehow

File: workspace/utilities/master/master.xsl (hunk 6 above)
Comment (dominic-blain) [reply workspace/utilities/master/master.xsl]: @nitriques Fixed!

File: workspace/utilities/com/site-nav.xsl [hunk 7]
```diff
@@ -85,11 +85,6 @@
 			<xsl:with-param name="attr" select="$attr" />
 			<xsl:with-param name="content">
 
-				<!-- nav-link (COMP: nav-link) -->
-				<xsl:call-template name="nav-link">
-					<xsl:with-param name="attr" select="$nav-link-attr"/>
-					<xsl:with-param name="selected-attr" select="$nav-link-selected-attr" />
-				</xsl:call-template>
 			</xsl:with-param>
```
Comment (nitriques) [inline workspace/utilities/com/site-nav.xsl]: I still fear that `util-nav-link-attr` won't get called. In understand that the link itself is porject-dependeant, but would it be possible to just output a link, with the call to `util-nav-link-attr` already made ?

File: workspace/utilities/lib/util-nav-link-attr.xsl [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+<!-- COMPONENT: util-nav-link-attr -->
+	<xsl:template name="util-nav-link-attr">
+		<xsl:param name="label" select="label" />
+		<xsl:param name="ga-cat" select="'menu-nav'" />
+		<xsl:param name="ga-label" select="concat($url-language, '/', $label)" />
```
Comment (nitriques) [inline workspace/utilities/lib/util-nav-link-attr.xsl]: Please add another parameter: `ga-action` . Its default value can be nothing.

File: workspace/utilities/com/site-nav.xsl (hunk 7 above)
Comment (dominic-blain) [reply workspace/utilities/com/site-nav.xsl]: Oh yeah absolutly. I was suggesting earlier to not make any changes in site nav for this PR and do it in another. But i figure I can still make some minimal adjustments to implement util-nav-link-attr in site-nav.

File: workspace/utilities/lib/util-nav-link-attr.xsl (hunk 8 above)
Comment (dominic-blain) [reply workspace/utilities/lib/util-nav-link-attr.xsl]: Done!

File: workspace/utilities/com/site-nav.xsl (hunk 7 above)
Comment (dominic-blain) [reply workspace/utilities/com/site-nav.xsl]: Done!

Comment (nitriques) [reply workspace/utilities/com/site-nav.xsl]: >  I was suggesting earlier

Yeah I did forgot about that sorry ;(

File: workspace/utilities/lib/util-nav-link-attr.xsl (hunk 8 above)
Comment (nitriques) [reply workspace/utilities/lib/util-nav-link-attr.xsl]: :+1:

**Comment by nitriques** [comment]
@PascalPiche if you could review this version, it would be cool. It changed a bit since you last checked it
